File: model/src/main/java/ru/mipt/acsl/decode/model/domain/DecodeUnit.java
package ru.mipt.acsl.decode.model.domain;

import org.jetbrains.annotations.NotNull;

import java.util.Optional;

/**
 * @author Artem Shein
 */
public interface DecodeUnit extends DecodeNameAware, DecodeOptionalInfoAware, DecodeReferenceable, DecodeNamespaceAware
{
    @NotNull
    Optional<String> getDisplay();

    @Override
    default <T> T accept(@NotNull DecodeReferenceableVisitor<T> visitor)
    {
        return visitor.visit(this);
    }
}

File: model/src/main/scala/ru/mipt/acsl/decode/model/domain/pure/registry/DecodeUnit.scala
package ru.mipt.acsl.decode.model.domain.pure.registry

import ru.mipt.acsl.decode.model.domain.{HasInfo, LocalizedString, NamespaceAware}
import ru.mipt.acsl.decode.model.domain.pure.Referenceable

/**
  * @author Artem Shein
  */
trait DecodeUnit extends HasInfo with Referenceable with NamespaceAware {
  def display: LocalizedString
}
